src/tools/Line.ts

```diff
@@ -0,0 +1,159 @@
+      layer.add(this.arrow)
+    } else if (this.endStyle === 'T') {
+      this.line = this.createLineElement(canvasElement)
+      this.TBar = this.createTElement(canvasElement)
+      layer.add(this.TBar)
+      layer.add(this.line)
+    }
+  }
+
+  // eslint-disable-next-line
+  mouseMoveAction = throttle((e: Konva.KonvaPointerEvent, canvasElement: CanvasElement, layer: Konva.Layer, socket: WebSocket): void => {
+    this.TBar.points([canvasElement.data[0], canvasElement.data[1], e.evt.x, e.evt.y])
+    if (this.endStyle === 'line') {
+      this.line.points([canvasElement.data[0], canvasElement.data[1], e.evt.x, e.evt.y])
+    } else if (this.endStyle === 'arrow') {
+      this.arrow.points([canvasElement.data[0], canvasElement.data[1], e.evt.x, e.evt.y])
+    } else if (this.endStyle === 'T') {
+      this.TBar.points(this.calcTBar(canvasElement.data[0], canvasElement.data[1], e.evt.x, e.evt.y))
+      this.line.points([canvasElement.data[0], canvasElement.data[1], e.evt.x, e.evt.y])
+    }
+    layer.batchDraw()
+  }, 5)
+
+  mouseUpAction = (e: Konva.KonvaPointerEvent, canvasElement: CanvasElement, _layer: Konva.Layer, socket: WebSocket): void => {
+    canvasElement.data = canvasElement.data.concat([e.evt.x, e.evt.y])
+    this.sendToWebsockets(socket, canvasElement)
+  }
+
+  renderCanvas = (canvasElement: CanvasElement, layer: Konva.Layer): void => {
+    this.strokeStyle = canvasElement.strokeStyle || 0
+    if (canvasElement.endStyle === 'line') {
+      layer.add(this.createLineElement(canvasElement, canvasElement.tool.colour, canvasElement.tool.size))
+    } else if (canvasElement.endStyle === 'arrow') {
+      layer.add(this.createArrowElement(canvasElement, canvasElement.tool.colour, canvasElement.tool.size))
+    } else if (canvasElement.endStyle === 'T') {
+      layer.add(this.createLineElement(canvasElement, canvasElement.tool.colour, canvasElement.tool.size))
+      layer.add(this.createTElement(canvasElement, canvasElement.tool.colour, canvasElement.tool.size).points(
+        this.calcTBar(canvasElement.data[0], canvasElement.data[1], canvasElement.data[2], canvasElement.data[3]))
+      )
+    } else {
+      layer.add(this.createLineElement(canvasElement, canvasElement.tool.colour, canvasElement.tool.size))
+    }
+    layer.add(this.createLineElement(canvasElement, canvasElement.tool.colour, canvasElement.tool.size))
+    layer.batchDraw()
+  }
+
+  createLineElement = (canvasElement: CanvasElement, colour?: string, size?: number): Konva.Shape & Konva.Line => {
+    return new Konva.Line({
+      globalCompositeOperation: 'source-over',
+      points: canvasElement.data,
+      stroke: colour || this.colour,
+      strokeWidth: size || this.size,
+      lineCap: 'mitter',
+      id: canvasElement.id,
+      dash: this.stroke[this.strokeStyle]
+    })
+  }
+
+  createTElement = (canvasElement: CanvasElement, colour?: string, size?: number): Konva.Shape & Konva.Line => {
```

Comment:
do you need these two functions to be separate? Can you not have one createLineElement? They look very similar